packages/@react-stately/virtualizer/src/Virtualizer.ts

```diff
@@ -624,13 +650,60 @@ export class Virtualizer<T extends object, V, W> {
   private _getLayoutInfoMap(rect: Rect, copy = false) {
     let layoutInfos = this.layout.getVisibleLayoutInfos(rect);
     let map = new Map;
+    let layoutInfosMap = new Map;
+    let persistedKeysMap = new Map;
 
     for (let layoutInfo of layoutInfos) {
       if (copy) {
         layoutInfo = layoutInfo.copy();
       }
 
-      map.set(layoutInfo.key, layoutInfo);
+      layoutInfosMap.set(layoutInfo.key, layoutInfo);
+    }
+
+    // Views that are outside of the viewable rectangle that we don't
+    // want to be reused by Virtualizer.
+    for (let persistKey of this._persistedKeys) {
+      let key = persistKey;
+
+      while (key) {
+        let layoutInfo = this.layout.getLayoutInfo(key);
+
+        if (layoutInfo) {
+          if (copy) {
+            layoutInfo = layoutInfo.copy();
+          }
+
+          persistedKeysMap.set(layoutInfo.key, layoutInfo);
+
+          // Assuming that any item with a parent is a subitem and we need to
+          // persist the parent layout, no parentKey is null
+          key = layoutInfo.parentKey;
+        } else {
+          key = null; // if there is no layoutInfo make sure the key is null to exit
+        }
+      }
+    }
+
+    // Merge layoutInfos and persistedKeys together in order of this._collection
+    let layoutInfosIterator = layoutInfosMap.keys();
+    let currentLayoutInfosKey = layoutInfosIterator.next().value;
+    let persistedKeyIterator = persistedKeysMap.keys();
+    let currentPersistedKey = persistedKeyIterator.next().value;
+    for (let collectionKey of this._collection.getKeys()) {
+      if (currentLayoutInfosKey && currentLayoutInfosKey === collectionKey) {
+        map.set(currentLayoutInfosKey, layoutInfosMap.get(currentLayoutInfosKey));
+        currentLayoutInfosKey = layoutInfosIterator.next().value;
+      } else if (currentPersistedKey && currentPersistedKey === collectionKey) {
+        map.set(currentPersistedKey, persistedKeysMap.get(currentPersistedKey));
+        currentPersistedKey = persistedKeyIterator.next().value;
+      }
+    }
+
+    // there are cases where we have layoutInfos (loader, placeholder, etc.) that aren't in the
+    // collection annd we need to make sure we persist those.
+    if (map.size !== (new Map([...layoutInfosMap, ...persistedKeysMap])).size) {
```

Comment:
this will iterate everything again, can simplify
```suggestion
    if (map.size !== layoutInfosMap.size + persistedKeysMap.size) {
```

Reply:
This won't work because a persistedKey can exist in both maps.

Reply:
ah, then should we be counting as we add them to the maps so we don't need to create copies of the Maps?

Reply:
that or change it to
let newMap = new Map([...layoutInfosMap, ...persistedKeysMap])
if (map.size !== newMap.size) map = newMap
?

Reply:
Your second code suggestion should work. Or I should find a better way to identify loaders and spinners (things not in the collection) and add them to the merged map.